code/modules/events/disease_outbreak.dm

```diff
@@ -267,25 +274,35 @@
 				/datum/symptom/fire,
 				/datum/symptom/flesh_death,
 				/datum/symptom/flesh_eating,
-				/datum/symptom/narcolepsy,
-				/datum/symptom/voice_change,
+				/datum/symptom/visionloss,
 			)
 
-	for(var/i in 1 to max_symptoms)
+	var/current_severity = 0
+
+	while(symptoms.len < max_symptoms)
 		var/datum/symptom/chosen_symptom = pick_n_take(possible_symptoms)
-		if(chosen_symptom)
-			var/datum/symptom/new_symptom = new chosen_symptom
-			symptoms += new_symptom
 
-	if(prob(50))
-		symptoms += new /datum/symptom/viralevolution
+		if(!chosen_symptom)
+			CRASH("Advanced disease could not pick a symptom!")
+
+		var/datum/symptom/new_symptom = new chosen_symptom
+
+		if(symptoms.len == (max_symptoms - 1) && (current_severity < requested_severity) && (new_symptom.severity < requested_severity))
```

Comment:
I feel like doing the whole `symptons.len == (max_symptoms - 1)` check should be at the top of this while loop, before we declare `new_symptom` variable as a type of early continue. Also, might be apt to do something like `symptoms.len >= max_symptoms`, not sure why you decrement max_symptoms by 1 for the evaluation, smells odd. Unsure though, but should incorporate some sort of early continue before we even start declaring variables.

Reply:
You're right, it should be checking if the symptom is valid before creating it.

For the second part, symptoms.len == (max_symptoms - 1) is to continue only if this is the last symptom we're trying to pick, though while moving it I realized the potential difference in severity by adding this is miniscule so it can be removed.